src/neo4j_graphrag/llm/openai_llm.py

```diff
@@ -72,20 +77,51 @@ def get_messages(
         input: str,
         message_history: Optional[Union[List[LLMMessage], MessageHistory]] = None,
         system_instruction: Optional[str] = None,
-    ) -> Iterable[ChatCompletionMessageParam]:
-        messages = []
+    ) -> Sequence[
+        ChatCompletionMessageParam
+    ]:  # Returns messages compatible with OpenAI's API
+        messages: List[ChatCompletionMessageParam] = []
         if system_instruction:
-            messages.append(SystemMessage(content=system_instruction).model_dump())
+            messages.append(
+                ChatCompletionSystemMessageParam(
+                    role="system", content=str(system_instruction)
+                )
+            )
         if message_history:
             if isinstance(message_history, MessageHistory):
                 message_history = message_history.messages
             try:
                 MessageList(messages=cast(list[BaseMessage], message_history))
             except ValidationError as e:
                 raise LLMGenerationError(e.errors()) from e
-            messages.extend(cast(Iterable[dict[str, Any]], message_history))
-        messages.append(UserMessage(content=input).model_dump())
-        return messages  # type: ignore
+            for msg in message_history:
+                if isinstance(msg, dict):
+                    role = msg.get("role")
+                    content = msg.get("content")
+                else:
+                    msg_dict = msg.model_dump()
+                    role = msg_dict.get("role")
+                    content = msg_dict.get("content")
+                if role == "system":
+                    messages.append(
+                        ChatCompletionSystemMessageParam(
+                            role="system", content=str(content)
+                        )
+                    )
+                elif role == "user":
+                    messages.append(
+                        ChatCompletionUserMessageParam(
+                            role="user", content=str(content)
+                        )
+                    )
+                elif role == "assistant":
+                    messages.append(
+                        ChatCompletionAssistantMessageParam(
+                            role="assistant", content=str(content)
+                        )
+                    )
+        messages.append(ChatCompletionUserMessageParam(role="user", content=str(input)))
+        return messages
```

Comment:
Not sure about this change, but mypy kept complaining locally so I had to do this. Happy to undo.